packages/image_picker/image_picker/ios/Classes/FLTPHPickerImageHandlingOperation.m

```diff
@@ -0,0 +1,132 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTPHPickerImageHandlingOperation.h"
+
+API_AVAILABLE(ios(14))
+@interface FLTPHPickerImageHandlingOperation ()
+
+@property(strong, nonatomic) PHPickerResult *result;
+@property(weak, nonatomic) NSMutableArray *pathList;
+@property(assign, nonatomic) NSNumber *maxHeight;
+@property(assign, nonatomic) NSNumber *maxWidth;
+@property(assign, nonatomic) NSNumber *desiredImageQuality;
+@property(assign, nonatomic) NSInteger index;
+
+@end
+
+@implementation FLTPHPickerImageHandlingOperation {
+  BOOL executing;
+  BOOL finished;
+}
+
+- (instancetype)initWithResult:(PHPickerResult *)result
+                      pathlist:(NSMutableArray *)pathList
+                     maxHeight:(NSNumber *)maxHeight
+                      maxWidth:(NSNumber *)maxWidth
+           desiredImageQuality:(NSNumber *)desiredImageQuality
+                         index:(NSInteger)index API_AVAILABLE(ios(14)) {
```

Comment:
I think it is better to give a result back instead of passing the array and index.
You can for example pass a result back like this: https://stackoverflow.com/a/19373929/2999543
That way this class is only responsible for getting the path and the calling class is responsible for managing a list.

Reply:
Good call! I think we can pass a block to the operation to get the result.

Reply:
Done.